src/app/[locale]/settings/_components/role-settings/RuleDialog.tsx

```diff
@@ -1,139 +1,178 @@
 import { Box, Button, Dialog, DialogActions, DialogContent, Typography } from '@mui/material';
 import { useTranslations } from 'next-intl';
-import { BaSyxRbacRule } from 'lib/services/rbac-service/types/RbacServiceData';
 import { useState } from 'react';
 import EditIcon from '@mui/icons-material/Edit';
 import { ArrowBack, Delete } from '@mui/icons-material';
 import { DialogCloseButton } from 'components/basics/DialogCloseButton';
 import { TargetInformationView } from 'app/[locale]/settings/_components/role-settings/target-information/TargetInformationView';
-import * as rbacActions from 'lib/services/rbac-service/RbacActions';
+import { deleteAndCreateRbacRule } from 'lib/services/rbac-service/RbacActions';
 import { mapFormModelToBaSyxRbacRule } from 'app/[locale]/settings/_components/role-settings/FormMappingHelper';
 import { useShowError } from 'lib/hooks/UseShowError';
 import { useNotificationSpawner } from 'lib/hooks/UseNotificationSpawner';
 import { RuleForm, RuleFormModel } from 'app/[locale]/settings/_components/role-settings/RuleForm';
+import { BaSyxRbacRule } from 'lib/services/rbac-service/types/RbacServiceData';
+import { CreateHint, DeleteHint } from 'app/[locale]/settings/_components/role-settings/HintDialogContent';
 import { RuleDeleteDialog } from 'app/[locale]/settings/_components/role-settings/RuleDeleteDialog';
 
+export type DialogRbacRule = BaSyxRbacRule & {
+    // If this rule is the only rule for the role
+    isOnlyRule: boolean;
+};
+
 type RuleDialogProps = {
-    readonly onClose: (reload: boolean) => Promise<void>;
+    readonly onClose: () => void;
+    readonly reloadRules: () => Promise<void>;
     readonly open: boolean;
-    readonly rule: BaSyxRbacRule;
+    readonly rule: DialogRbacRule;
+    readonly availableRoles: string[];
 };
 
-export const RuleDialog = (props: RuleDialogProps) => {
+type DialogMode = 'edit' | 'view' | 'delete' | 'create-hint' | 'delete-hint';
+
+export const RuleDialog = ({ onClose, reloadRules, open, rule, availableRoles }: RuleDialogProps) => {
     const t = useTranslations('pages.settings.rules');
-    const [mode, setMode] = useState<'edit' | 'view' | 'delete'>('view');
+    const [dialogMode, setDialogMode] = useState<DialogMode>('view');
     const { showError } = useShowError();
     const notificationSpawner = useNotificationSpawner();
 
+    async function afterSubmit(newData: RuleFormModel) {
+        const isNewRole = !availableRoles.includes(newData.role);
+        const isDeletedRole = rule.isOnlyRule && rule.role !== newData.role;
```

Comment:
```suggestion
    async function afterUpdate(newData: RuleFormModel) {
        const isRuleForNewRole = !availableRoles.includes(newData.role);
        const lastRuleForRoleDeleted = rule.isOnlyRuleForRole && rule.role !== newData.role;
```

The Submit should also be update. It was a bit hard to read for me. isOnlyRuleForRole is optional :wink:

Reply:
I find this a bit hard to understand in general. onSubmit() calls afterSubmit() which calls onClose() and reloadRules()

Reply:
maybe it's just the name afterSubmit() which confuses me. Maybe call the method with a more speaking name of what it does

Reply:
changed it to `updateDialogModeForNewRule` and added two small comments